src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_INDEX_FOR_FIGURE = 5;
+    private static final int MAX_INDEX_FOR_PARAM = 100;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randIndexForFigure = random.nextInt(MAX_INDEX_FOR_FIGURE);
+        switch (randIndexForFigure) {
+            case 0:
+                double radiusCircle = random.nextInt(MAX_INDEX_FOR_PARAM) + 1;
+                String colorCircle = colorSupplier.getRandomColor();
+                return new Circle(radiusCircle, colorCircle);
+            case 1:
+                double firstBaseIsTr = random.nextInt(MAX_INDEX_FOR_PARAM) + 1;
+                double secondBaseIsTr = random.nextInt(MAX_INDEX_FOR_PARAM) + 1;
+                double heightIsTr = random.nextInt(MAX_INDEX_FOR_PARAM) + 1;
+                String colorIsTr = colorSupplier.getRandomColor();
+                return new IsoscelesTrapezoid(firstBaseIsTr, secondBaseIsTr, heightIsTr, colorIsTr);
```

Comment:
Please extract code within case block to separate methods for better readability